src/resolver.rs

```diff
@@ -85,9 +92,112 @@ impl Resolver {
+                            trace!("Processing NLA: {:?}", nla);
+                            if let GenlCtrlAttrs::McastGroups(groups) = nla {
+                                trace!("Found McastGroups: {:?}", groups);
+                                for group in groups {
+                                    // 'group' is a Vec<McastGrpAttrs>
+                                    let mut group_name = None;
+                                    let mut group_id = None;
+    
+                                    for group_attr in group {
+                                        trace!("Processing group_attr: {:?}", group_attr);
+                                        match group_attr {
+                                            McastGrpAttrs::Name(ref name) => {
+                                                group_name = Some(name.clone());
+                                                trace!("Found group name: '{}'", name);
+                                            }
+                                            McastGrpAttrs::Id(id) => {
+                                                group_id = Some(id);
+                                                trace!("Found group id: {}", id);
+                                            }
+                                        }
+                                    }
+    
+                                    if let (Some(name), Some(id)) = (group_name, group_id) {
+                                        mc_groups.insert(name.clone(), id);
+                                        trace!(
+                                            "Inserted group '{}' with id {} into mc_groups",
+                                            name,
+                                            id
+                                        );
+                                    }
+                                }
+                            } else {
+                                trace!("Unhandled NLA: {:?}", nla);
+                            }
+                        }
+                    }
+                    NetlinkPayload::Error(e) => {
+                        error!("Received NetlinkPayload::Error: {:?}", e);
+                        return Err(e.into());
+                    }
+                    other => {
+                        warn!("Received unexpected NetlinkPayload: {:?}", other);
+                    }
+                }
+            }
+            trace!("Finished processing responses");
+    
+            // Update the cache
+            self.groups_cache.insert(family_name, mc_groups.clone());
+            trace!("Updated groups_cache for family_name: '{}'", family_name);
+    
+            trace!("Returning mc_groups: {:?}", mc_groups);
+            Ok(mc_groups)
+        }
+    }
+    
+
     pub fn clear_cache(&mut self) {
         self.cache.clear();
+        self.groups_cache.clear();
     }
+
 }
 
 #[cfg(all(test, tokio_socket))]
```

Comment:
The tests are actually never run and always disabled ... 

```suggestion
#[cfg(all(test, feature = "tokio_socket"))]
```